### PR TITLE
feat: introduce sqllogictest for dozer

### PR DESCRIPTION
**Pull request by xudong963**
Introduce sqllogictest for dozer, make it more flexible to add massive tests.

Ideally, the test set of sqlite can be reused later!

### CODE REVIEW

File: dozer-tests/Cargo.toml [hunk 1]
```diff
@@ -25,6 +29,16 @@ futures = { version = "0.3.26", optional = true }
 env_logger = "0.10.0"
 clap = { version = "4.1.6", features = ["derive"] }
 dozer-orchestrator = { path = "../dozer-orchestrator" }
+rusqlite = { version = "0.28.0", features = ["bundled", "column_decltype"] }
+sqllogictest = "0.11.1"
+sqlparser = "0.32.0"
```
Comment (v3g42) [inline dozer-tests/Cargo.toml:34]: Can you re-export sqlparser from `dozer-sql` ?

File: dozer-tests/Cargo.toml [hunk 2]
```diff
@@ -25,6 +29,16 @@ futures = { version = "0.3.26", optional = true }
...
+thiserror = "1.0.38"
```
Comment (v3g42) [inline dozer-tests/Cargo.toml:35]: This is already available through `dozer_types::thiserror`

File: dozer-tests/Cargo.toml [hunk 3]
```diff
@@ -14,6 +14,10 @@ path = "src/dozer_tests.rs"
 name = "dozer-test-client"
 path = "src/dozer_test_client.rs"
 
+[[bin]]
+name = "dozer-sqllogictest"
+path = "src/sqllogictest/src/main.rs"
+
 [dependencies]
```
Comment (v3g42) [inline dozer-tests/Cargo.toml:21]: Considering we are building a binary, can we restrict the dependencies with a certain feature flag ? So we can build it separately ?

**Review by v3g42 — CHANGES_REQUESTED** [review]

**Comment by v3g42** [comment]
Seems like a good idea. But it would be good to see some of the existing test cases ported before we adopt it widely.

**Review by snork-alt — CHANGES_REQUESTED** [review]
Good start. But if we develop a new framework we need support for complex SQLs. On top of this, all previous test cases should be ported to the new framework.

For complex scenarios you can refer to this : https://github.com/getdozer/dozer-samples/blob/main/pg-flights/dozer-config.yaml

File: dozer-tests/src/sqllogictest/suits/join.test [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+2 jack nicholson
+3 angelina jolie
+3 angelina jolie
+3 angelina jolie
+3 luigi guiness
+3 luigi guiness
+3 luigi guiness
+4 tom hanks
+
+# dozer has bug for the sql, less one row `1 penelope guiness`
```
Comment (xudong963) [inline dozer-tests/src/sqllogictest/suits/join.test:278]: I find a left join bug  @mediuminvader

Comment (mediuminvader) [reply dozer-tests/src/sqllogictest/suits/join.test:278]: Nice, but I'm refactoring the join right now #1180 , let's re run this on the next merge

File: dozer-tests/Cargo.toml (hunk 3 above)
Comment (xudong963) [reply dozer-tests/Cargo.toml:21]: It doesn't need many dependencies, so maybe we don't need to use a certain feature flag to restrict it

**Review by v3g42 — APPROVED** [review]
Thanks @xudong963. This looks good to me. 

I believe @mediuminvader wanted to incorporate a few more changes. i ll leave it you and him.

**Comment by xudong963** [comment]
So let's merge？ Then let's iterate together? @mediuminvader 

I still have some ideas to solid new framework.

**Review by snork-alt — APPROVED** [review]

**Pull request merged**